core/src/main/java/bisq/core/support/dispute/refund/RefundManager.java

```diff
@@ -238,4 +252,62 @@ public void onDisputeResultMessage(DisputeResultMessage disputeResultMessage) {
     }
+
+    public CompletableFuture<List<Transaction>> requestBlockchainTransactions(String makerFeeTxId,
+                                                                              String takerFeeTxId,
+                                                                              String depositTxId,
+                                                                              String delayedPayoutTxId) {
+        NetworkParameters params = btcWalletService.getParams();
+        List<Transaction> txs = new ArrayList<>();
+        return mempoolService.requestTxAsHex(makerFeeTxId)
+                .thenCompose(txAsHex -> {
+                    txs.add(new Transaction(params, Hex.decode(txAsHex)));
+                    return mempoolService.requestTxAsHex(takerFeeTxId);
+                }).thenCompose(txAsHex -> {
+                    txs.add(new Transaction(params, Hex.decode(txAsHex)));
+                    return mempoolService.requestTxAsHex(depositTxId);
+                }).thenCompose(txAsHex -> {
+                    txs.add(new Transaction(params, Hex.decode(txAsHex)));
+                    return mempoolService.requestTxAsHex(delayedPayoutTxId);
+                })
+                .thenApply(txAsHex -> {
+                    txs.add(new Transaction(params, Hex.decode(txAsHex)));
+                    return txs;
+                });
+    }
+
+    public void verifyTradeTxChain(List<Transaction> txs) {
+        Transaction makerFeeTx = txs.get(0);
+        Transaction takerFeeTx = txs.get(1);
+        Transaction depositTx = txs.get(2);
+        Transaction delayedPayoutTx = txs.get(3);
+
+        // The order and number of buyer and seller inputs are not part of the trade protocol consensus.
+        // In the current implementation buyer inputs come before seller inputs at depositTx and there is
+        // only 1 input per trader, but we do not want to rely on that.
+        // So we just check that both fee txs are found in the inputs.
+        boolean makerFeeTxFoundAtInputs = false;
+        boolean takerFeeTxFoundAtInputs = false;
+        for (TransactionInput transactionInput : depositTx.getInputs()) {
+            String fundingTxId = transactionInput.getOutpoint().getHash().toString();
+            if (!makerFeeTxFoundAtInputs) {
+                makerFeeTxFoundAtInputs = fundingTxId.equals(makerFeeTx.getTxId().toString());
+            }
+            if (!takerFeeTxFoundAtInputs) {
+                takerFeeTxFoundAtInputs = fundingTxId.equals(takerFeeTx.getTxId().toString());
+            }
+        }
+        checkArgument(makerFeeTxFoundAtInputs, "makerFeeTx not found at depositTx inputs");
+        checkArgument(takerFeeTxFoundAtInputs, "takerFeeTx not found at depositTx inputs");
+        checkArgument(depositTx.getInputs().size() >= 2,
+                "DepositTx must have at least 2 inputs");
+        checkArgument(depositTx.getOutputs().size() == 1,
+                "DepositTx must have 1 input");
```

Comment:
output

Reply:
Yes that was wrong. I removed the check in the last commit.